File: web-server-cms/src/main/java/com/pigatron/web/cms/content/service/ContentService.java
package com.pigatron.web.cms.content.service;

import com.pigatron.web.cms.content.entity.Content;
import com.pigatron.web.cms.content.repository.ContentRepository;
import com.pigatron.web.core.service.AbstractRepositoryService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class ContentService extends AbstractRepositoryService<Content> {

    @Autowired
    private ContentRepository contentRepository;

    public ContentService() {
        super(Content.class);
    }

    public Optional<Content> getPageByUrlKey(String urlKey) {
        return contentRepository.findPageByUrlKey(urlKey);
    }

    public Optional<Content> getPublishedPageByUrlKey(String urlKey) {
        return contentRepository.findPublishedPageByUrlKey(urlKey);
    }
}

File: web-admin-cms/src/main/resources/static/admin/cms/service/content.service.ts
import {Inject, Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {AbstractDataService} from "web-admin-core/main";
import {Content} from "./content";

@Injectable()
export class ContentService extends AbstractDataService<Content> {

    constructor(@Inject(HttpClient) http: HttpClient) {
        super(http, 'api/cms/content');
    }

}